### PR TITLE
fix(nano): ecdsa policy binding support for encrypt

### PR DESCRIPTION
**Pull request by sujankota**

### CODE REVIEW

**Comment by sujankota** [comment]
Working on unit tests

File: lib/src/nanotdf/encrypt.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ import {
   exportCryptoKey,
 } from '../nanotdf-crypto/index.js';
 import { KasPublicKeyInfo } from '../access.js';
+import {
+  computeECDSASig,
+  extractRSValuesFromSignature,
+} from 'src/nanotdf-crypto/ecdsaSignature.js';
```
Comment (dmihalcik-virtru) [inline lib/src/nanotdf/encrypt.ts:22]:
```suggestion
} from '../nanotdf-crypto/ecdsaSignature.js';
```

File: lib/src/nanotdf-crypto/ecdsaSignature.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * @param {Uint8Array} signature - The signature to be verified.
+ * @param {Uint8Array} data - The data that was signed.
+ * @returns {Promise<boolean>} - A promise that resolves to a boolean indicating whether the signature is valid.
+ */
+export async function verifyECDSASignature(
+  publicKey: CryptoKey,
+  signature: Uint8Array,
+  data: Uint8Array
+): Promise<boolean> {
+  const isValid = await window.crypto.subtle.verify(
```
Comment (dmihalcik-virtru) [inline lib/src/nanotdf-crypto/ecdsaSignature.ts:42]:
```suggestion
  const isValid = await crypto.subtle.verify(
```

File: lib/src/index.ts [hunk 3]
```diff
@@ -104,13 +114,17 @@ export class NanoTDFClient extends Client {
...
-   * @param data to decrypt
+   * @param {string | TypedArray | ArrayBuffer} data - The data to be encrypted.
+   * @param {EncryptOptions} [options=defaultOptions] - The encryption options (currently unused).
+   * @returns {Promise<ArrayBuffer>} A promise that resolves to the encrypted data as an ArrayBuffer.
+   * @throws {Error} If the initialization vector is not a number.
    */
-  async encrypt(data: string | TypedArray | ArrayBuffer): Promise<ArrayBuffer> {
+  async encrypt(
+    data: string | TypedArray | ArrayBuffer,
+    options: EncryptOptions = defaultOptions
```
Comment (dmihalcik-virtru) [inline lib/src/index.ts:126]:
```suggestion
    options?: EncryptOptions
```

**Review by dmihalcik-virtru — APPROVED** [review]

**Review by dmihalcik-virtru — DISMISSED** [review]

**Review by dmihalcik-virtru — APPROVED** [review]

**Pull request merged**